src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift

```diff
@@ -39,9 +40,18 @@ enum Route {
 				  testInformation.timestamp >= 0
 			else {
 				self = .rapidAntigen( .failure(.invalidTestCode))
+				Log.error("Antigen test data is not, either timeStamp is -ve or the hash is invalid", log: .qrCode)
+				return
+			}
+			let allIsNil = testInformation.firstName == nil && testInformation.firstName == nil && testInformation.dateOfBirthString == nil
```

Comment:
I would rather check for `isEmpty`:
```suggestion
			let allIsNil = testInformation.firstName.isEmpty && testInformation.firstName.isEmpty && testInformation.dateOfBirthString.isEmpty
```